### PR TITLE
support virtualservice multi port

### PR DESCRIPTION
**Pull request by zackzhangkai**
Signed-off-by: zackzhangkai <zhangkaiamm@gmail.com>

**What type of PR is this?**

If a service has multi port, the traffic will always be routed to the first one. So we should support multi port service.

https://github.com/kubesphere/kubesphere/issues/3428

### CODE REVIEW

File: pkg/controller/utils/servicemesh/application.go [hunk 1]
```diff
@@ -135,7 +136,18 @@ func FillDestinationPort(vs *v1alpha3.VirtualService, service *corev1.Service) {
...
+				vs.Spec.Tcp[i].Match[j].Port = uint32(service.Spec.Ports[0].Port)
 			}
 		}
 	}
 }
+
+func SupportHttpProtocol(protocol string) bool {
+	httpPro := []string{"http", "http2", "grpc"}
+	for _, i := range httpPro {
+		if protocol == i || strings.HasPrefix(protocol, i+"-") {
```
Comment (zryfish) [inline pkg/controller/utils/servicemesh/application.go:148]: case sensitive match?

Comment (zackzhangkai) [reply pkg/controller/utils/servicemesh/application.go:148]: https://istio.io/latest/docs/ops/configuration/traffic-management/protocol-selection/  shows supporting lower case

Comment (zryfish) [reply pkg/controller/utils/servicemesh/application.go:148]: https://github.com/istio/istio/blob/master/pkg/config/protocol/instance.go#L61 Please take a look at this, and I think we should add some test cases

Comment (zackzhangkai) [reply pkg/controller/utils/servicemesh/application.go]: Fixed

**Pull request merged**